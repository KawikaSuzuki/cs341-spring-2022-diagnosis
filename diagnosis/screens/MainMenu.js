import React from 'react';
import { Text, View, StyleSheet, Image, TouchableOpacity } from 'react-native';
import { ExtraText } from '../components/styles';

var time = new Date().getHours();

var button1Cliked = () => {
    alert("Button 1 clicked");
}

var button2Cliked = () => {
    alert("Button 2 clicked");
}

var button3Cliked = () => {
    alert("Button 3 clicked");
}

var button4Cliked = () => {
    alert("Button 4 clicked");
}

var button5Cliked = () => {
    alert("Button 5 clicked");
}

const MainMenu = () => (

    
        <View style={styles.mainContainer}> 
            <View style={styles.headerContainer}>   
                <Image style={styles.PDXLogo} source={require('diagnosis/assets/simplePDXLogo.png')}></Image>
                <Image style={styles.accountIcon} source={require('diagnosis/assets/accountIcon.png')}></Image>
            </View>
            
            <Image style={styles.stethoscopeImage} source={require('diagnosis/assets/stethoscope.jpeg')} blurRadius={1}></Image>
            <Text style={styles.welcomeText}>Good Morning, {"\n"}
                User
            </Text>

            <View style={styles.middleLineLeft}></View>
            <ExtraText style={styles.PDXCenterText}>PDX</ExtraText>
            <View style={styles.middleLineRight}></View>

            <View style={styles.quickLinksView}>
                <TouchableOpacity style={styles.quickLinkButtons} onPress={button1Cliked}> 
                    <Text style={styles.buttonText}>Knowledge Test</Text>    
                </TouchableOpacity>

                <TouchableOpacity style={styles.quickLinkButtons} onPress={button2Cliked}> 
                    <Text style={styles.buttonText}>Topic of the Day</Text>
                </TouchableOpacity>

                <TouchableOpacity style={styles.quickLinkButtons} onPress={button3Cliked}> 
                    <Text style={styles.buttonText}>Community</Text>
                </TouchableOpacity>

                <TouchableOpacity style={styles.quickLinkButtons} onPress={button4Cliked}> 
                    <Text style={styles.buttonText}>Our Team</Text>
                </TouchableOpacity>

                <TouchableOpacity style={styles.quickLinkButtons} onPress={button5Cliked}> 
                    <Text style={styles.buttonText}>About Us</Text>
                </TouchableOpacity>
            </View>
        </View>
);

// style sheet to keep the enhancements clean and readable
const styles = StyleSheet.create({
    mainContainer: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
        backgroundColor: "#BFAA8C",
    },

    headerContainer: {
        backgroundColor: "#ffffff",
        position: "absolute",
        top: 0,
        height: 100,
        justifyContent: "center",
        alignContent: "center",
        width: '100%',
    },

    PDXLogo: {
        resizeMode: "contain",
        width: 200,
        top: 15,
        left: 90,
        flex: 1,
    },

    accountIcon: {
        position: "absolute",
        resizeMode: "contain",
        width: 50,
        height: 50,
        right: 15,
        top: 40
    },

    stethoscopeImage: {
        resizeMode: "contain",
        position: "absolute",
        width: '100%',
        height: undefined,
        aspectRatio: 1,
        top: 38
    },

    welcomeText: {
        position: "absolute",
        color: "#E63610",
        fontSize: 60,
        top: 120,
        fontWeight: 'bold',
        textAlign: "center",
        fontFamily: "Gill Sans",
        textShadowColor:'#585858',
        textShadowOffset:{width: 5, height: 5},
        textShadowRadius:10,
    },
    
    quickLinksView: {
        top: 70,
        backgroundColor: "white",
        justifyContent: 'center',
        position: "relative",
        marginTop: 280,
        height: 370,
        marginBottom: 0,
        width: '96%',
        borderRadius: 60,
        borderColor: 'red',
        alignContent: 'center'

    },

    magGlassImg: {
        resizeMode: "contain",
        width: '15%',
        position: 'relative'
    },

    quickLinkButtons: {
        marginTop: 15,
        alignItems: 'center',
        marginLeft: 50,
        width: '75%',
        paddingVertical: 10,
        backgroundColor: 'red',
        borderRadius: 30,
    },
    
    middleLineLeft: {
        width: 100,
        borderBottomColor: 'grey',
        borderBottomWidth: 6,
        borderRadius: 12,
        position: 'absolute',
        left: 20,
        top: 390
    },

    PDXCenterText: {
        position: 'absolute',
        fontSize: 50,
        color: 'white',
        textShadowColor:'#585858',
        textShadowOffset:{width: 1, height: 9},
        textShadowRadius:10,
        paddingBottom: 30,
        left: 145,
        fontWeight: 'bold'
    },

    middleLineRight: {
        width: 100,
        borderBottomColor: 'grey',
        borderBottomWidth: 6,
        borderRadius: 12,
        position: 'absolute',
        right: 20,
        top: 390
    },

    buttonText: {
        fontSize: 25,
        fontFamily: "Gill Sans",
        fontWeight: "bold",
        color: "black"
    }
});
    
export default MainMenu; // send it off



