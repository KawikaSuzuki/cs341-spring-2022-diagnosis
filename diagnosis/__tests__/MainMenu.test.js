import React from 'react';
import renderer from 'react-test-renderer';
import {act, fireEvent, render, waitFor} from '@testing-library/react-native';
import MainMenu from '../screens/MainMenu';

describe('<MainMenu />', () => {

    test('renders correctly', () => { // test screen rendering
        const tree = renderer.create(<MainMenu/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('test button 1', function() { // test button one links correctly
        const navigate = jest.fn();
        const {queryByTestId} = render(<MainMenu navigation={{navigate}}/>);
        const buttonOne = queryByTestId('buttonOne');
        expect(buttonOne).toBeTruthy();
        fireEvent.press(buttonOne);
        expect(navigate).toHaveBeenCalledWith('KnowledgeTest');        
    });

    test('test button 2', function() { // test button two links correctly
        const navigate = jest.fn();
        const {queryByTestId} = render(<MainMenu navigation={{navigate}}/>);
        const buttonTwo = queryByTestId('buttonTwo');
        expect(buttonTwo).toBeTruthy();
        fireEvent.press(buttonTwo);
        expect(navigate).toHaveBeenCalledWith('Lectures');        
    });

    test('test button 3', function() { // test button three links correctly
        const navigate = jest.fn();
        const {queryByTestId} = render(<MainMenu navigation={{navigate}}/>);
        const buttonThree = queryByTestId('buttonThree');
        expect(buttonThree).toBeTruthy();
        fireEvent.press(buttonThree);
        expect(navigate).toHaveBeenCalledWith('Community');        
    });

    test('test button 4', function() { // test button four links correctly
        const navigate = jest.fn();
        const {queryByTestId} = render(<MainMenu navigation={{navigate}}/>);
        const buttonFour = queryByTestId('buttonFour');
        expect(buttonFour).toBeTruthy();
        fireEvent.press(buttonFour);
        expect(navigate).toHaveBeenCalledWith('TeamInfo');        
    });

    test('test button 5', function() { // test button five links correctly
        const navigate = jest.fn();
        const {queryByTestId} = render(<MainMenu navigation={{navigate}}/>);
        const buttonFive = queryByTestId('buttonFive');
        expect(buttonFive).toBeTruthy();
        fireEvent.press(buttonFive);
        expect(navigate).toHaveBeenCalledWith('AboutUs');        
    });


});