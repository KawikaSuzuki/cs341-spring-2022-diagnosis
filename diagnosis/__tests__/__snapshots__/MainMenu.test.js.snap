// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenu /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#BFAA8C",
      "flex": 1,
      "height": 1334,
      "justifyContent": "center",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "height": 166.75,
        "justifyContent": "center",
        "position": "absolute",
        "top": 0,
        "width": 750,
      }
    }
  >
    <View>
      <Text />
    </View>
    <Image
      source={1}
      style={
        Object {
          "left": 87,
          "position": "absolute",
          "resizeMode": "contain",
          "top": -60,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "position": "absolute",
          "resizeMode": "contain",
          "right": 15,
          "top": 40,
          "width": 50,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "height": 250,
        "position": "absolute",
        "top": 100,
        "width": 750,
      }
    }
  >
    <Image
      blurRadius={1}
      source={1}
      style={
        Object {
          "flex": 1,
          "height": 1334,
          "position": "relative",
          "resizeMode": "stretch",
          "width": 750,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#E63610",
          "fontFamily": "Gill Sans",
          "fontSize": 60,
          "fontWeight": "bold",
          "left": 55,
          "position": "absolute",
          "textAlign": "center",
          "textShadowColor": "#585858",
          "textShadowOffset": Object {
            "height": 5,
            "width": 5,
          },
          "textShadowRadius": 10,
          "top": 20,
        }
      }
    >
      Good Morning, 
      

      User
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 80,
        "position": "absolute",
        "top": 350,
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "grey",
          "borderBottomWidth": 6,
          "borderRadius": 12,
          "left": 20,
          "position": "absolute",
          "top": 42,
          "width": 100,
        }
      }
    />
    <Text
      className="sc-egiyK eqlRhB"
      style={
        Object {
          "color": "white",
          "fontSize": 50,
          "fontWeight": "bold",
          "left": 140,
          "position": "absolute",
          "textShadowColor": "#585858",
          "textShadowOffset": Object {
            "height": 9,
            "width": 1,
          },
          "textShadowRadius": 10,
          "top": 17,
        }
      }
    >
      PDX
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "grey",
          "borderBottomWidth": 6,
          "borderRadius": 12,
          "position": "absolute",
          "right": 20,
          "top": 42,
          "width": 100,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "red",
        "borderRadius": 60,
        "padding": 10,
        "position": "absolute",
        "top": 440,
        "width": "96%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 30,
          "margin": 10,
          "opacity": 1,
          "paddingVertical": 10,
          "width": "75%",
        }
      }
      testID="buttonOne"
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Gill Sans",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        Knowledge Test
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 30,
          "margin": 10,
          "opacity": 1,
          "paddingVertical": 10,
          "width": "75%",
        }
      }
      testID="buttonTwo"
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Gill Sans",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        Topic of the Day
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 30,
          "margin": 10,
          "opacity": 1,
          "paddingVertical": 10,
          "width": "75%",
        }
      }
      testID="buttonThree"
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Gill Sans",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        Community
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 30,
          "margin": 10,
          "opacity": 1,
          "paddingVertical": 10,
          "width": "75%",
        }
      }
      testID="buttonFour"
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Gill Sans",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        Our Team
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 30,
          "margin": 10,
          "opacity": 1,
          "paddingVertical": 10,
          "width": "75%",
        }
      }
      testID="buttonFive"
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Gill Sans",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        About Us
      </Text>
    </View>
  </View>
</View>
`;
